packages/account/src/Components/address-details/address-details.tsx

```diff
@@ -7,176 +16,183 @@ import {
+ * @param {boolean} is_gb_residence - is residence Great Britan
+ * @param {Function} onSubmitEnabledChange - function to change submit button status
+ * @param {React.RefObject} selected_step_ref - reference to selected step
+ * @param {Function} fetchStatesList - function to fetch states list
+ * @param {Object} value - form values
+ * @param {Array} disabled_items - array of disabled fields
+ * @param {boolean} has_real_account - has real account
+ * @returns {React.ReactNode} - returns react node
+ */
+const AddressDetails = observer(
+    ({
+        getCurrentStep,
+        onSave,
+        onCancel,
+        goToNextStep,
+        goToPreviousStep,
+        validate,
+        onSubmit,
+        is_svg,
+        is_mf,
+        is_gb_residence,
+        onSubmitEnabledChange,
+        selected_step_ref,
+        disabled_items,
+        has_real_account,
+        ...props
+    }: TAddressDetails) => {
+        const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
+        const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
 
-const AddressDetails = ({
-    states_list,
-    getCurrentStep,
-    onSave,
-    onCancel,
-    goToNextStep,
-    goToPreviousStep,
-    validate,
-    onSubmit,
-    is_svg,
-    is_mf,
-    is_gb_residence,
-    onSubmitEnabledChange,
-    selected_step_ref,
-    disabled_items,
-    has_real_account,
-    ...props
-}: TAddressDetails) => {
-    const { is_appstore } = React.useContext(PlatformContext);
-    const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
-    const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
+        const { client, ui } = useStore();
 
-    React.useEffect(() => {
-        const { cancel, promise } = makeCancellablePromise(props.fetchStatesList());
-        promise.then(() => {
-            setHasFetchedStatesList(true);
-            if (props.value.address_state) {
-                setAddressStateToDisplay(getLocation(states_list, props.value.address_state, 'text'));
-            }
-        });
-        return () => {
-            setHasFetchedStatesList(false);
-            cancel();
-        };
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+        const { states_list, fetchStatesList } = client;
+        const { is_desktop, is_mobile } = ui;
 
-    const is_submit_disabled_ref = React.useRef<boolean | undefined>(true);
+        React.useEffect(() => {
+            const { cancel, promise } = makeCancellablePromise(fetchStatesList());
+            promise.then(() => {
+                setHasFetchedStatesList(true);
+                if (props.value.address_state) {
+                    setAddressStateToDisplay(getLocation(states_list, props.value.address_state, 'text'));
+                }
+            });
+            return () => {
+                setHasFetchedStatesList(false);
+                cancel();
+            };
+        }, []);
```

Comment:
I suspect the dependency array should've some values

Reply:
I have removed the business logic out of the component and moved it to a hook. useEffect is removed now